core/sql/common/NAMemory.h

```diff
@@ -468,6 +468,11 @@ NA_EIDPROC
   NAMemoryType getType() {  return type_; }
 NA_EIDPROC
   NABoolean getUsage(size_t* lastSegSize, size_t* freeSize, size_t* totalSize);
+  // for debugging
+NA_EIDPROC
+  NABoolean containsAddress(void *addr)
+        { return NABlock::blockHolding(firstBlk_, addr) != NULL; }
+
```

Comment:
Yes, I added the following code to the GroupAttributes constructor to find this leak, out of thousands of allocations it did:

>  const char *loc;
> 
>  if ((long) this < 0x8000000)
>    loc = "c++h";
>  else if ((long) this > 0x7ff000000000)
>    loc = "stck";
>  else if (CmpCommon::statementHeap() == NULL)
>    loc = "nost";
>  else if (CmpCommon::statementHeap()->containsAddress(this))
>    loc = "stmt";
>  else if (CmpCommon::contextHeap()->containsAddress(this))
>    loc = "ctxt";
>  else
>    loc = "unkn";
>
>  printf("GroupAttributes1, %#018lx %s\n", (long) this, loc);